src/cmps/tags/Tags.tsx

```diff
@@ -1,51 +1,92 @@
+// import { string, number } from 'prop-types';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faPlus } from '@fortawesome/free-solid-svg-icons';
+// import { any, element } from 'prop-types';
 
+library.add(faPlus);
 
 class Tags extends Component {
-
     state = {
-        tags: [{
-            id: 1,
-            name: 'tag1'
-        }, {
-            id: 2,
-            name: 'tag2'
-        }]
-    }
+        currentTag: '',
+        tags: [],
+    };
 
-    remove = () => {
-        console.log('button X clicked');
-    }
+    remove = (id: string) => {
+        let index = this.state.tags.findIndex((element: any) => element.id === id);
+        this.setState(this.state.tags.splice(index, 1));
+    };
+
+    handleSubmit = (e: any) => {
+        e.preventDefault();
+        let isDouble = this.checkDouble(this.state.currentTag);
+        if (isDouble) {
+            let title = this.state.currentTag;
+            let id = this.makeId(9);
+            if (title.length <= 0) return;
+            this.setState({
+                currentTag: '',
+                tags: [...this.state.tags, { title, id }],
+            });
+        } else {
+            this.setState({
+                currentTag: '',
+            });
+        }
+    };
 
-    addTag = (event: any) => {
-        event.preventDefault();
-        // let tag: string = this.refs.tag.value;
-        // let id: string = makeId();
-        // console.log(this.refs.tag.value);
-        // this.setState.
+    checkDouble(tag: string) {
+        let arr: any[] = [];
+        this.state.tags.forEach((element: any) => {
+            arr.push(element.title);
+        });
+        arr.push(tag);
+        let uniqueItems = [...new Set(arr)];
+        return uniqueItems.length === arr.length;
     }
 
-    makeId(): string {
-        return 'xxx'
+    makeId(length: number): string {
+        var text = '';
+        var possible = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
+        for (var i = 0; i < length; i++) text += possible.charAt(Math.floor(Math.random() * possible.length));
+        return text;
     }
 
     render() {
         return (
             <section>
                 <div className="tags-container">
-                    <Tag value="Tag1" isDeletable={true} onClick={this.remove} />
-                    <form onSubmit={this.addTag.bind(this)}>
-                        <input type="text" ref="tag" placeholder="Enter tag name" />
-
+                    <form className="tags-form" onSubmit={e => this.handleSubmit(e)}>
+                        <input
+                            type="text"
+                            placeholder="Add a tag"
+                            className="tags-input"
+                            value={this.state.currentTag}
+                            onChange={e => this.setState({ currentTag: e.target.value })}
```

Comment:
extract to class method `handleChange`

same reason as above

Reply:
as above I did not understand ... and I did not find a better way ...